x/register/genesis.go

```diff
@@ -12,79 +12,74 @@ import (
+		if err != nil {
+			panic(err)
+		}
+		switch resourceNode.GetStatus() {
+		case stakingtypes.Bonded:
 			lenOfGenesisBondedResourceNode++
 			initialStakeTotal = initialStakeTotal.Add(resourceNode.Tokens)
-			initResourceNodeBondedPoolBalances = initResourceNodeBondedPoolBalances.Add(sdk.Coin{
-				Denom:  keeper.BondDenom(ctx),
-				Amount: resourceNode.Tokens,
-			})
-		} else if resourceNode.GetStatus() == stakingtypes.Unbonded {
-			initResourceNodeNotBondedPoolBalances = initResourceNodeNotBondedPoolBalances.Add(sdk.Coin{
-				Denom:  keeper.BondDenom(ctx),
-				Amount: resourceNode.Tokens,
-			})
+			if freshStart {
+				err = keeper.SendCoinsFromAccountToResNodeBondedPool(ctx, ownerAddr, sdk.NewCoin(keeper.BondDenom(ctx), resourceNode.Tokens))
+				if err != nil {
+					panic(err)
+				}
+			}
+		case stakingtypes.Unbonded:
+			if freshStart {
+				err = keeper.SendCoinsFromAccountToResNodeNotBondedPool(ctx, ownerAddr, sdk.NewCoin(keeper.BondDenom(ctx), resourceNode.Tokens))
+				if err != nil {
+					panic(err)
+				}
+			}
+		default:
+			panic(types.ErrInvalidNodeStat)
 		}
 		keeper.SetResourceNode(ctx, resourceNode)
 	}
 	// set initial genesis number of resource nodes
 	keeper.SetBondedResourceNodeCnt(ctx, sdk.NewInt(lenOfGenesisBondedResourceNode))
-	err := keeper.MintResourceNodeNotBondedPoolWhenInitGenesis(ctx, initResourceNodeNotBondedPoolBalances)
-	if err != nil {
-		panic(err)
-	}
-	err = keeper.MintResourceNodeBondedPoolWhenInitGenesis(ctx, initResourceNodeBondedPoolBalances)
-	if err != nil {
-		panic(err)
-	}
 
 	lenOfGenesisBondedMetaNode := int64(0)
-	initMetaNodeNotBondedPoolBalances := sdk.NewCoins(sdk.Coin{
-		Denom:  keeper.BondDenom(ctx),
-		Amount: sdk.ZeroInt(),
-	})
-	initMetaNodeBondedPoolBalances := sdk.NewCoins(sdk.Coin{
-		Denom:  keeper.BondDenom(ctx),
-		Amount: sdk.ZeroInt(),
-	})
 	for _, metaNode := range data.GetMetaNodes() {
-		if metaNode.GetStatus() == stakingtypes.Bonded {
+		ownerAddr, err := sdk.AccAddressFromBech32(metaNode.OwnerAddress)
+		if err != nil {
+			panic(err)
+		}
+		switch metaNode.GetStatus() {
+		case stakingtypes.Bonded:
 			lenOfGenesisBondedMetaNode++
 			initialStakeTotal = initialStakeTotal.Add(metaNode.Tokens)
-			initMetaNodeBondedPoolBalances = initMetaNodeBondedPoolBalances.Add(sdk.Coin{
-				Denom:  keeper.BondDenom(ctx),
-				Amount: metaNode.Tokens,
-			})
-		} else if metaNode.GetStatus() == stakingtypes.Unbonded {
-			initMetaNodeNotBondedPoolBalances = initMetaNodeNotBondedPoolBalances.Add(sdk.Coin{
-				Denom:  keeper.BondDenom(ctx),
-				Amount: metaNode.Tokens,
-			})
+			if freshStart {
+				err = keeper.SendCoinsFromAccountToMetaNodeBondedPool(ctx, ownerAddr, sdk.NewCoin(keeper.BondDenom(ctx), metaNode.Tokens))
+				if err != nil {
+					panic(err)
+				}
+			}
+		case stakingtypes.Unbonded:
+			if freshStart {
+				err = keeper.SendCoinsFromAccountToMetaNodeBondedPool(ctx, ownerAddr, sdk.NewCoin(keeper.BondDenom(ctx), metaNode.Tokens))
```

Comment:
shouldn't this be send to unbonded pool?

Reply:
Yep sry for the typo